### PR TITLE
Skenny/document download speed

### PR DESCRIPTION
**Pull request by goldilocked**

### CODE REVIEW

File: src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs [hunk 1]
```diff
@@ -286,37 +288,229 @@ int IWaterAllocationAccessor.GetWaterRightsCount(WaterRightsSearchCriteria searc
...
+                    .Add(regulatoryUUID.RegulatoryOverlayUuid);
+                return ValueTask.CompletedTask;
+            });
+            return allRegulatoryUuids;
+        }
+
+        private async Task<ConcurrentDictionary<long, ConcurrentBag<string>>> GetWaterSourcesForSites(WaterRightsSearchCriteria searchCriteria)
+        {
+            var (db, sites) = GetFilteredSites(searchCriteria);
+            var matchingWaterSourceUUIDs = db.WaterSourceBridgeSitesFact
```
Comment (goldilocked) [inline src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs:501]: Double check that distincts are no longer needed for all queries

File: src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs [hunk 2]
```diff
@@ -286,37 +288,229 @@ int IWaterAllocationAccessor.GetWaterRightsCount(WaterRightsSearchCriteria searc
...
+                .Where(a => sites.Any(b => b.SiteId == a.SiteId))
```
Comment (goldilocked) [inline src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs:502]: Add asnotracking

File: src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs [hunk 3]
```diff
@@ -286,37 +288,229 @@ int IWaterAllocationAccessor.GetWaterRightsCount(WaterRightsSearchCriteria searc
...
+                .Select(a => new { a.SiteId, a.WaterSource.WaterSourceUuid })
```
Comment (goldilocked) [inline src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs:503]: Add text to empty csv files, along the lines of "no data found for sites"
Talk with Adel and Ryan

File: src/API/WesternStatesWater.WestDaat.Tests.ManagerTests/WaterAllocationManagerTests.cs [hunk 4]
```diff
@@ -702,37 +649,44 @@ public async Task WaterRightsAsZip_BuildsStream_AllCollectionsWithData()
...
+                        {
+                            case "Organizations.csv":
+                                await CheckRecords(zipFile, zipFile.FileName, organizations);
+                                break;
+                            case "Methods.csv":
+                                await CheckRecords(zipFile, zipFile.FileName, methods);
+                                break;
+                            case "Variables.csv":
+                                await CheckRecords(zipFile, zipFile.FileName, variables);
+                                break;
```
Comment (goldilocked) [inline src/API/WesternStatesWater.WestDaat.Tests.ManagerTests/WaterAllocationManagerTests.cs:674]: Add a unit test that checks if the tuples returned from the accessor have the correct name/data combinations

File: src/API/WesternStatesWater.WestDaat.Managers/WaterAllocationManager.cs [hunk 5]
```diff
@@ -194,27 +195,28 @@ public void WaterRightsAsZip(Stream responseStream, ClientContracts.WaterRightsS
...
-                        var entry = new ZipEntry(ZipEntry.CleanName($"{file.GetType().GetGenericArguments()[0].Name}.csv"));
+                        var entry = new ZipEntry(ZipEntry.CleanName($"{file.Name}.csv"));
                         zipStream.PutNextEntry(entry);
 
                         ms.Seek(0, SeekOrigin.Begin);
 
                         StreamUtils.Copy(ms, zipStream, new byte[4096]);
                     }
-                }
+                    Console.WriteLine($"finished {file.Name}");
```
Comment (goldilocked) [inline src/API/WesternStatesWater.WestDaat.Managers/WaterAllocationManager.cs:218]: Remove console.writelines

File: src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs [hunk 6]
```diff
@@ -286,37 +288,229 @@ int IWaterAllocationAccessor.GetWaterRightsCount(WaterRightsSearchCriteria searc
...
+
+            await foreach (var site in sites)
+            {
+                site.WaterSourceUuids = string.Join(",", allWaterSourceUUIDS.GetValueOrDefault(site.SiteId) ?? new ConcurrentBag<string>());
+                site.RegulatoryOverlayUuids = string.Join(",", allRegulatory.GetValueOrDefault(site.SiteId) ?? new ConcurrentBag<string>());
+                yield return site.Map<CsvModels.Sites>();
+            }
+        }
+
+        internal class WaterAllocationsHelper: CsvModels.WaterAllocations
```
Comment (goldilocked) [inline src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs:333]: Move helper classes to their own files

File: src/API/WesternStatesWater.WestDaat.Tests.ManagerTests/WaterAllocationManagerTests.cs (hunk 4 above)
Comment (goldilocked) [reply src/API/WesternStatesWater.WestDaat.Tests.ManagerTests/WaterAllocationManagerTests.cs:674]: Attempted for a while to create this test but was unable to find a way to loop over results of accessor in test in thread safe way

**Review by EverettRhodes — APPROVED** [review]

**Pull request merged**